src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const extraData = action.extraData;
+    const keysToRemove = action.keysToRemove;
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in extraData) {
+          state[key] = extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const value of keysToRemove) {
+          delete state[value];
+        }
```

Comment:
Let's create separate function for this, to make `switch-case` smaller and more readable